swarm/storage/encryption/encryption_test.go

```diff
@@ -0,0 +1,125 @@
+
+var hashFunc = sha3.NewKeccak256
+
+func TestEncryptDataLongerThanPadding(t *testing.T) {
+	enc := New(4095, uint32(0), hashFunc)
+
+	data := make([]byte, 4096)
+	key := make([]byte, 32)
+
+	_, err := enc.Encrypt(data, key)
+	if err == nil || !strings.Contains(err.Error(), "Data length longer than padding") {
+		t.Fatalf("Expected error \"Data length longer than padding\" got %v", err)
+	}
+}
+
+func TestEncryptDataZeroPadding(t *testing.T) {
+	enc := New(0, uint32(0), hashFunc)
+
+	data := make([]byte, 2048)
+	key := make([]byte, 32)
+
+	encrypted, err := enc.Encrypt(data, key)
+	if err != nil {
+		t.Fatalf("Expected no error got %v", err)
+	}
+	if len(encrypted) != 2048 {
+		t.Fatalf("Encrypted data length expected %v got %v", 2048, len(encrypted))
+	}
+}
+
+func TestEncryptDataLengthEqualsPadding(t *testing.T) {
+	enc := New(4096, uint32(0), hashFunc)
+
+	data := make([]byte, 4096)
+	key := make([]byte, 32)
+
+	encrypted, err := enc.Encrypt(data, key)
+	if err != nil {
+		t.Fatalf("Expected no error got %v", err)
+	}
+	encryptedHex := common.Bytes2Hex(encrypted)
+	expectedTransformed := common.Hex2Bytes(expectedTransformedHex)
+
+	if !bytes.Equal(encrypted, expectedTransformed) {
+		t.Fatalf("Expected %v got %v", expectedTransformedHex, encryptedHex)
+	}
+}
+
+func TestEncryptDataLengthSmallerThanPadding(t *testing.T) {
+	enc := New(4096, uint32(0), hashFunc)
+
+	data := make([]byte, 4080)
+	key := make([]byte, 32)
+
+	encrypted, err := enc.Encrypt(data, key)
+	if err != nil {
+		t.Fatalf("Expected no error got %v", err)
+	}
+	if len(encrypted) != 4096 {
+		t.Fatalf("Encrypted data length expected %v got %v", 4096, len(encrypted))
+	}
+}
+
+func TestEncryptDataCounterNonZero(t *testing.T) {
+	// TODO
+}
+
+func TestDecryptDataLengthNotEqualsPadding(t *testing.T) {
+	enc := New(4096, uint32(0), hashFunc)
+
+	data := make([]byte, 4097)
+	key := make([]byte, 32)
+
+	_, err := enc.Decrypt(data, key)
+	if err == nil || !strings.Contains(err.Error(), "Data length different than padding") {
+		t.Fatalf("Expected error \"Data length different than padding\" got %v", err)
+	}
+}
+
+func TestEncryptDecryptIsIdentity(t *testing.T) {
```

Comment:
use `rand.Read` to test many. and create benchmark for both ways.

Reply:
Sure, these are not all the tests I wanted to write, just the beginning